File: src/test/java/net/sf/rails/game/state/OwnerImpl.java
package net.sf.rails.game.state;

import net.sf.rails.game.state.AbstractItem;
import net.sf.rails.game.state.Item;
import net.sf.rails.game.state.Owner;

/**
 * Implementation of an Owner used for Testing
 */

class OwnerImpl extends AbstractItem implements Owner {
    
    private OwnerImpl(Item parent, String id) {
        super(parent, id);
    }

    static OwnerImpl create(Item parent, String id) {
        return new OwnerImpl(parent, id);
    }
    
    public String toString() {
        return "Owner(Id=" + getId() + ")";
    }

    
}

File: dotnet/GameStateTest/OwnerImpl.cs
using GameLib.Net.Game.State;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/**
 * Implementation of an Owner used for Testing
 */

namespace GameStateTest
{
    public class OwnerImpl : AbstractItem, IOwner
    {
        private OwnerImpl(IItem parent, string id) : base(parent, id)
        {
        }

        public static OwnerImpl Create(IItem parent, String id)
        {
            return new OwnerImpl(parent, id);
        }

        override public string ToString()
        {
            return "Owner(Id=" + Id + ")";
        }
    }
}
